src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java

```diff
@@ -147,4 +151,78 @@ public void validatesGoodYaml() throws Exception {
             )
         );
     }
+
+    /**
+     * Verifies that execute throws an exception, if there is no .ebextensions
+     * directory in the WAR file.
+     * @throws Exception Thrown in case of error.
+     */
+    @Test
+    public void checkEbextensionsValidityThrowsExceptionNoDir()
+        throws Exception {
+        // @checkstyle IllegalTypeCheck (2 lines)
+        final AbstractBeanstalkMojo mojo =
+            Mockito.mock(AbstractBeanstalkMojo.class);
+        final File mockFile = Mockito.mock(File.class);
+        final ZipFile mockZipFile = Mockito.mock(ZipFile.class);
+        Mockito.when(mockFile.exists()).thenReturn(true);
+        Mockito.when(mojo.createZipFile()).thenReturn(mockZipFile);
+        Mockito.doCallRealMethod().when(mojo).execute();
+        Mockito.doCallRealMethod().when(mojo)
+            .setWar(Mockito.any(File.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .validate(Mockito.any(ZipFile.class));
+        mojo.setWar(mockFile);
+        Mockito.when(mockZipFile.getEntry(".ebextensions")).thenReturn(null);
+        try {
+            mojo.execute();
+            Assert.fail("Expect MojoFailureException to be thrown");
+        } catch (final MojoFailureException exception) {
+            MatcherAssert.assertThat(
+                exception.getMessage(),
+                Matchers.equalTo(
+                    ".ebextensions directory does not exist in the WAR file"
+                )
+            );
+        }
+    }
+
+    /**
+     * Verifies that execute throws an exception, if the .ebextensions directory
+     * is empty.
+     * @throws Exception Thrown in case of error.
+     */
+    @Test
+    @SuppressWarnings("unchecked")
+    public void checkEbextensionsValidityThrowsExceptionNoConfigFiles()
+        throws Exception {
+        // @checkstyle IllegalTypeCheck (2 lines)
+        final AbstractBeanstalkMojo mojo =
+            Mockito.mock(AbstractBeanstalkMojo.class);
+        Mockito.doCallRealMethod().when(mojo).execute();
+        Mockito.doCallRealMethod().when(mojo).createZipFile();
+        Mockito.doCallRealMethod().when(mojo)
+            .setWar(Mockito.any(File.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .validate(Mockito.any(ZipFile.class));
```

Comment:
@mkordas Sure, so not this test, but a common part of the code in these tests establishes an `AbstractBeanstalkMojo`, `final File mockFile = Mockito.mock(File.class);` and `final ZipFile mockZipFile = Mockito.mock(ZipFile.class);` Each on one line, I can't refactor out one method for these 3 lines, best I could do would be 3. Then it goes on to wire up the mocks, more methods. In test fixture, this could look like:

    new AbstractBeanstalkMojoFixture()
	    .WithMockFileAndZipFile();